### PR TITLE
Add API documentation to all services

### PR DESCRIPTION
**Pull request by mistydemeo**
This also removes some unused functions from the `ArchivesSpace` service and performs a few other miscellaneous cleanups I ran into while going over the services.

### CODE REVIEW

File: app/services/facet.service.js [hunk 1]
```diff
@@ -2,7 +2,24 @@ import angular from 'angular';
...
+//              the object has a value at that key which is an exact match.
+// * regexp   - if the test value is a regexp, then the facet matches if
+//              the object has a value at that key which passes value.match(regexp).
+// * function - if the test value is a function, then the facet matches if
+//              the function passes true when passed the value at that key from the object.
 factory('Facet', function() {
+  // Adds a new facet.
+  // * name - the key this function should match.
+  // * value - the value to match against the object's value; see above.
+  // * data
```
Comment (Hwesta) [inline app/services/facet.service.js]: What's `data`?

File: app/services/tag.service.js [hunk 2]
```diff
@@ -3,25 +3,29 @@ import 'restangular';
...
+  var get = function(id) {
     return Tag.one(id).one('tags').getList();
   };
 
+  // Updates the list of tags associated with the file with the specified `uuid`.
   var submit = function(id, tags) {
     Tag.one(id).one('tags').customPUT(tags).then(null, response => {
       // TODO display error handling
-      $log.error('Submitting tags for file ' + String(id) + ' failed with response: ', response.status);
+      $log.error(`Submitting tags for file ${String(id)} failed with response: ${response.status}`);
```
Comment (Hwesta) [inline app/services/tag.service.js:21]: Should template strings be documented in the Babel section?

File: app/services/transfer.service.js [hunk 3]
```diff
@@ -64,10 +66,26 @@ factory('Transfer', ['Alert', 'Facet', 'Restangular', 'Tag', function(Alert, Fac
...
+    // This provides a more convenient method to retrieve files by ID without needing
+    // to go through a getter function.
+    // This is populated when `resolve` is called.
     id_map: {},
+    // A flat list of all unique tags associated with the files in this transfer.
+    // This is initially populated by `resolve`, and is updated whenever tags are
+    // added or removed using the methods on this service.
     tags: [],
+    // Provided a set of data retrieved from the transfer backlog, performs the following:
+    // * populates the `data`, `formats` and `id_map` attributes.
```
Comment (Hwesta) [inline app/services/transfer.service.js]: Also populates `tags`? Or should the tags comment be updated?

File: app/services/facet.service.js (hunk 1 above)
Comment (mistydemeo) [inline app/services/facet.service.js]: I had a hard time remembering this. :sweat_smile: I'll look it up and document it.

Comment (mistydemeo) [inline app/services/facet.service.js]: Okay, documented.

File: app/services/tag.service.js (hunk 2 above)
Comment (mistydemeo) [inline app/services/tag.service.js:21]: I'll add it in a new PR.

File: app/services/transfer.service.js (hunk 3 above)
Comment (mistydemeo) [inline app/services/transfer.service.js]: Fixed the comment.

**Comment by Hwesta** [comment]
:+1:

**Pull request merged**